app/services/customers/create_service.rb

```diff
@@ -176,7 +176,13 @@ def handle_api_billing_configuration(customer, params, new_customer)
 
       return if customer.payment_provider.nil?
 
-      create_or_update_provider_customer(customer, billing)
+      update_provider_customer = (billing || {})[:provider_customer_id].present?
+      update_provider_customer ||=
+        customer.__send__("#{customer.payment_provider}_customer")&.provider_customer_id.present?
+
+      if billing.key?(:payment_provider) && update_provider_customer
```

Comment:
Style/GuardClause: Use a guard clause (`return unless billing.key?(:payment_provider) && update_provider_customer`) instead of wrapping the code inside a conditional expression. (https://rubystyle.guide#no-nested-conditionals)